mobx/test/reaction_test.dart

```diff
@@ -35,6 +35,40 @@ void main() {
           executed, isFalse); // reaction has been disposed, so no more effects
     });
 
+    group('equals override', () {
+      test('basics work', () {
+        var executed = false;
+
+        bool equals(_, __) => false;
+
+        final x = Observable(10, equals: equals);
+        final d = reaction(
+          (_) => x.value,
+          (_) {
+            executed = true;
+          },
+          name: 'Basic Reaction',
+          equals: equals,
+        );
+
+        expect(executed, isFalse);
+
+        x.value = 11;
+        expect(executed, isTrue);
```

Comment:
Not sure how this is working as the comparator is always returning false. Hence the effect function should never execute?

Reply:
It's the other way around, the original implementation had a case for _not_ running when the items are _equal_, that was an optimization to not run extra functions for the same values.

Reply:
Ah ok. Got it. The snippet out of context confused me a bit 🤔.